pynecone/state.py

```diff
@@ -257,6 +260,11 @@ def _set_default_value(cls, prop: BaseVar):
             field.required = False
             field.default = default_value
 
+    @classmethod
+    @functools.lru_cache()
```

Comment:
This shouldn't be a classmethod since it applies to the instance of the state, can just be:
```
def get_current_page(self) -> str:
    return self.router_data["pathname"]
```

This also needs a docstring.